pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -1328,19 +1336,12 @@ func compareCAPIInfraMachineTemplates(platform configv1.PlatformType, infraMachi
 			return nil, errAssertingCAPIAWSMachineTemplate
 		}
 
-		diff := make(map[string]any)
-
-		if diffSpec := deep.Equal(typedInfraMachineTemplate1.Spec, typedinfraMachineTemplate2.Spec); len(diffSpec) > 0 {
-			diff[".spec"] = diffSpec
-		}
-
-		if diffObjectMeta := util.ObjectMetaEqual(typedInfraMachineTemplate1.ObjectMeta, typedinfraMachineTemplate2.ObjectMeta); len(diffObjectMeta) > 0 {
-			diff[".metadata"] = diffObjectMeta
-		}
-
-		// TODO: Evaluate if we want to add status comparison if needed in the future (e.g. for scale from zero capacity).
-
-		return diff, nil
+		spec1 = typedInfraMachineTemplate1.Spec
+		spec2 = typedinfraMachineTemplate2.Spec
+		metadata1 = typedInfraMachineTemplate1.ObjectMeta
```

Comment:
A minor readability thing, this grouping is a little strange to me. I would have expected something like


```
spec1 = ...
metadata1 = ...
status1 = ...

spec2 = ...
metadata2 = ...
status2 = ...
```

or 

```
spec1 = ...
spec2 = ...

status1 = ...
status2 = ...

metadata1 = ...
metadata2 = ...
```